### PR TITLE
[Speech] Add support for Xcode 15

### PR DESCRIPTION
**Pull request by haritha-mohan**

### CODE REVIEW

File: src/speech.cs [hunk 1]
```diff
@@ -49,6 +49,14 @@ public enum SFSpeechRecognizerAuthorizationStatus : long {
 		Authorized,
 	}
 
+	[Mac(14, 0), iOS(17, 0), MacCatalyst(17,0)]
```
Comment (dalexsoto) [inline src/speech.cs]: Nit
```suggestion
	[Mac (14, 0), iOS (17, 0), MacCatalyst (17,0)]
```

File: src/speech.cs [hunk 2]
```diff
@@ -347,4 +359,55 @@ interface SFVoiceAnalytics : NSCopying, NSSecureCoding {
 		[Export ("voicing", ArgumentSemantic.Copy)]
 		SFAcousticFeature Voicing { get; }
 	}
+
+	[Mac(14, 0), iOS(17, 0), MacCatalyst(17, 0)]
```
Comment (dalexsoto) [inline src/speech.cs]:
```suggestion
	[Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
```

File: src/speech.cs [hunk 3]
```diff
@@ -347,4 +359,55 @@ interface SFVoiceAnalytics : NSCopying, NSSecureCoding {
...
+	[BaseType(typeof(NSObject))]
```
Comment (dalexsoto) [inline src/speech.cs]:
```suggestion
	[BaseType (typeof (NSObject))]
```

File: src/speech.cs [hunk 4]
```diff
@@ -347,4 +359,55 @@ interface SFVoiceAnalytics : NSCopying, NSSecureCoding {
...
+	interface SFSpeechLanguageModelConfiguration : NSCopying {
+		[Export("initWithLanguageModel:")]
+		NativeHandle Constructor(NSUrl languageModel);
```
Comment (dalexsoto) [inline src/speech.cs]:
```suggestion
		NativeHandle Constructor (NSUrl languageModel);
```

File: src/speech.cs [hunk 5]
```diff
@@ -347,4 +359,55 @@ interface SFVoiceAnalytics : NSCopying, NSSecureCoding {
...
+
+		[Export("initWithLanguageModel:vocabulary:")]
+		NativeHandle Constructor(NSUrl languageModel, [NullAllowed] NSUrl vocabulary);
```
Comment (dalexsoto) [inline src/speech.cs]:
```suggestion
		NativeHandle Constructor (NSUrl languageModel, [NullAllowed] NSUrl vocabulary);
```

File: src/speech.cs [hunk 6]
```diff
@@ -347,4 +359,55 @@ interface SFVoiceAnalytics : NSCopying, NSSecureCoding {
...
+
+		[Export("languageModel", ArgumentSemantic.Copy)]
+		NSUrl LanguageModel { get; }
+
+		[NullAllowed, Export("vocabulary", ArgumentSemantic.Copy)]
+		NSUrl Vocabulary { get; }
+	}
+
+	[Mac (14,0), iOS (17,0), MacCatalyst(17, 0)]
```
Comment (dalexsoto) [inline src/speech.cs]:
```suggestion
	[Mac (14,0), iOS (17,0), MacCatalyst (17, 0)]
```

File: src/speech.cs [hunk 7]
```diff
@@ -347,4 +359,55 @@ interface SFVoiceAnalytics : NSCopying, NSSecureCoding {
...
+	[BaseType (typeof(NSObject))]
```
Comment (dalexsoto) [inline src/speech.cs]:
```suggestion
	[BaseType (typeof (NSObject))]
```

File: src/speech.cs [hunk 8]
```diff
@@ -347,4 +359,55 @@ interface SFVoiceAnalytics : NSCopying, NSSecureCoding {
...
+		[Field("SFAnalysisContextTagSelectedText")]
+		NSString SelectedText { get; }
+	}
+
+	[Partial]
+	[Mac(14, 0), iOS(17, 0), MacCatalyst(17,0)]
+	interface SFSpeech {
+		[Field("SFSpeechErrorDomain")]
+		NSString ErrorDomain { get; }
+	}
```
Comment (dalexsoto) [inline src/speech.cs]: Remove this and use the [ErrorDomain ("SFSpeechErrorDomain")] attribute on top of `SFSpeechErrorCode`
```suggestion
```

File: src/speech.cs [hunk 9]
```diff
@@ -49,6 +49,14 @@ public enum SFSpeechRecognizerAuthorizationStatus : long {
 		Authorized,
 	}
 
+	[Mac(14, 0), iOS(17, 0), MacCatalyst(17,0)]
+	[Native]
+	public enum SFSpeechErrorCode : long {
```
Comment (dalexsoto) [inline src/speech.cs:54]:
```suggestion
	[Native]
	[ErrorDomain ("SFSpeechErrorDomain")]
	public enum SFSpeechErrorCode : long {
```

**Review by dalexsoto — CHANGES_REQUESTED** [review]
Small nits here and there

**Comment by dalexsoto** [comment]
ahh the bot did beat me in some cases 🤣

File: src/speech.cs [hunk 10]
```diff
@@ -347,4 +359,55 @@ interface SFVoiceAnalytics : NSCopying, NSSecureCoding {
...
+		[Field ("SFAnalysisContextTagSelectedText")]
+		NSString SelectedText { get; }
+	}
+
+	[Partial]
+	[Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	interface SFSpeech {
+		[Field ("SFSpeechErrorDomain")]
+		NSString ErrorDomain { get; }
+	}
```
Comment (haritha-mohan) [inline src/speech.cs:412]:
```suggestion
```

Comment (haritha-mohan) [inline src/speech.cs:412]:
```suggestion
```

**Comment by haritha-mohan** [comment]
> ahh the bot did beat me in some cases 🤣

haha, addressed the error domain comment

**Review by dalexsoto — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**